src/soar_sdk/cli/package/utils.py

```diff
@@ -1,46 +1,43 @@
 import httpx
+from contextlib import asynccontextmanager
+from collections.abc import AsyncGenerator
 
 
+@asynccontextmanager
 async def phantom_get_login_session(
     base_url: str, username: str, password: str
-) -> httpx.AsyncClient:
-    """Log into phantom and return the authenticated client. The caller needs to close the client."""
-    client = httpx.AsyncClient(base_url=base_url, verify=False)  # noqa: S501
-
-    # get the cookies from the get method
-    response = await client.get("/login")
-    csrf_token = response.cookies.get("csrftoken")
-
-    await client.post(
-        "/login",
-        data={
-            "username": username,
-            "password": password,
-            "csrfmiddlewaretoken": csrf_token,
-        },
-        cookies=response.cookies,
-        headers={"Referer": f"{base_url}/login"},
-    )
-
-    return client
+) -> AsyncGenerator[httpx.AsyncClient, None]:
+    """Contextmanager that creates an authenticated client with CSRF token handling."""
+    async with httpx.AsyncClient(base_url=base_url, verify=False) as client:  # noqa: S501
+        # get the cookies from the get method
+        response = await client.get("/login")
+        csrf_token = response.cookies.get("csrftoken")
+        client.cookies.update(response.cookies)
+
+        await client.post(
+            "/login",
+            data={
+                "username": username,
+                "password": password,
+                "csrfmiddlewaretoken": csrf_token,
+            },
+            headers={"Referer": f"{base_url}/login"},
+        )
+
+        yield client
 
 
 async def phantom_post(
```

Comment:
I just realized this method has a very generic name but pretty explicitly will only work for endpoints which expect to receive file contents in byte form, which is a tiny minority of endpoints you might POST to on Phantom. Might as well name it as such
```suggestion
async def phantom_install_app(
```